net_transport.go

```diff
@@ -793,19 +808,25 @@ func decodeResponse(conn *netConn, resp interface{}) (bool, error) {
 func sendRPC(conn *netConn, rpcType uint8, args interface{}) error {
 	// Write the request type
 	if err := conn.w.WriteByte(rpcType); err != nil {
-		conn.Release()
+		if err := conn.Release(); err != nil {
+			return err
+		}
```

Comment:
What's a circumstance where we're going to get an error here and want that to be the return value instead of the original error? (Similar for all these changes.)